### ISSUE TITLE
Shipping rate doesn't include tax when tax is configured to be included in prices

### ISSUE THREAD

**Issue by duncanmcclean** [damcclean/simple-commerce#856]
### Description

When you have tax configured to be included in prices, it works fine for products but tax isn't included in the shipping rate applied to the cart.

A workaround for this issue would be to ensure that the price returned by your shipping method includes tax.

```php
'tax_engine' => TaxEngine::class,
 
'tax_engine_config' => [
    // Basic Engine
    'rate'               => 10,
    'included_in_prices' => true,

    // Standard Tax Engine
    'address' => 'billing',

    'behaviour' => [
        'no_address_provided' => 'default_address',
        'no_rate_available' => 'prevent_checkout',
    ],

    'default_address' => [
        'address_line_1' => '',
        'address_line_2' => '',
        'city' => '',
        'region' => '',
        'country' => '',
        'zip_code' => '',
    ],
],
```

### Steps to reproduce

1. Setup a shipping method with a fixed tax rate
2. Configure tax so it's included in prices
3. See that the shipping price applied to the cart doesn't have tax added onto it

### Environment

Unsure - reported via email.

<sub>[DTD-53](https://linear.app/doublethreedigital/issue/DTD-53/shipping-rate-doesnt-include-tax-when-tax-is-configured-to-be-included)</sub>

**Comment by duncanmcclean** [comment]
This is a little more complicated than initially thought, especially for the 'standard' tax engine, as we'd need some way to figure out which of the tax rates you have setup relate to shipping.

With products, you can assign tax categories to product entries, then the rates are decided based on that & the customer's location. I need to figure out the best way to configure the category to use (maybe Simple Commerce will create a tax category just for shipping?).

**Comment by mscruse** [comment]
I think that could be good for an immediate solution as I know changing tax would be difficult. Is it possible to just add something to the shipping config for the tax rate?




[https://assets.mity.com.au/signature/michael-scruse.jpg]

Michael Scruse
Head of Arts & Crafts, Mity Digital
03 9012 7566<tel:03%209012%207566>

|       www.mity.com.au <https://www.mity.com.au>



[https://assets.mity.com.au/signature/social-facebook.png] <http://facebook.com/mitydigital>    [https://assets.mity.com.au/signature/social-linkedin.png]  <http://au.linkedin.com/in/michaelscruse>   [https://assets.mity.com.au/signature/social-twitter.png]  <http://twitter.com/mitydigital>     [https://assets.mity.com.au/signature/social-instagram.png]  <http://www.instagram.com/mitydigital>     Book a chat with me ***@***.***/bookings/s/GBAo46rWQkC5Jfi5nPZqoA2>




From: Duncan McClean ***@***.***>
Date: Sunday, 26 March 2023 at 05:24
To: duncanmcclean/simple-commerce ***@***.***>
Cc: Michael Scruse ***@***.***>, Manual ***@***.***>
Subject: Re: [duncanmcclean/simple-commerce] Shipping rate doesn't include tax when tax is configured to be included in prices (Issue #856)

This is a little more complicated than initially thought, especially for the 'standard' tax engine, as we'd need some way to figure out which of the tax rates you have setup relate to shipping.

With products, you can assign tax categories to product entries, then the rates are decided based on that & the customer's location. I need to figure out the best way to configure the category to use (maybe Simple Commerce will create a tax category just for shipping?).

**Comment by duncanmcclean** [comment]
This issue is still at the back of my mind - I've been busy the last couple of weeks with client work & getting Simple Commerce 5 ready to go.

I'll loop back to this once v5 has been released.
